core/src/com/unciv/models/tilesets/TileSetCache.kt

```diff
@@ -3,24 +3,52 @@ package com.unciv.models.tilesets
 import com.badlogic.gdx.Gdx
 import com.badlogic.gdx.files.FileHandle
 import com.unciv.JsonParser
+import com.unciv.UncivGame
 import com.unciv.ui.utils.ImageGetter
 
-object TileSetCache : HashMap<String, TileSetConfig>(){
-    fun loadTileSetConfigs(consoleMode: Boolean = false, printOutput: Boolean = false){
+object TileSetCache : HashMap<String, TileSetConfig>() {
+    private data class TileSetAndMod(val tileSet: String, val mod: String)
+    private val allConfigs = HashMap<TileSetAndMod, TileSetConfig>()
+
+    /** Combine [TileSetConfig]s for chosen mods.
+     * Vanilla always active, even with a base ruleset mod active.
+     * A mod with a mod name matching the currently chosen tileset is used if in permanent visual mods.
+     *    Note the mod name must exactly match the [TileSet] name with `" Tileset"` appended.
+     * Other active mods can be passed in parameter [ruleSetMods], if that is `null` and a game is in
+     * progress, that game's mods are used instead.
+     */
+    fun assembleTileSetConfigs(ruleSetMods: HashSet<String>? = null) {
+        val mods = mutableSetOf("")
+        if (UncivGame.isCurrentInitialized()) {
+            val settings = UncivGame.Current.settings
+            if (settings.tileSet + " Tileset" in settings.visualMods) mods.add(settings.tileSet + " Tileset")
```

Comment:
Not sure why this is needed.
If you want a tileset to be available, make the mod permanently visible and that's it. We shouldn't have to determine what miss the tiles come from.

Reply:
That's the idea - we no longer mix all TileSetConfigs together because then we got rules from DeCiv while the atlas from DeCiv isn't loaded which causes the observed render errors. When a mod is requested in GameInfo - merge that TileSetConfig, because the atlas will be active too. When a mod is permavisual we still need a way to merge the TileSetConfig when it's needed...

This may be a knot in my brain... What does marking 5Hex as permavisual but then _not_ selecting it in options actually do? Is merging the ruleVariants in that case good or bad? As I wrote that I thought 'bad' - but since the atlas is in and the merging is per TileSet - the actual answer is 'good'?

Also, my pinglish fails at your uses of "miss" - twice - may be some simple typo I don't get. If it was something important, please choose different words.

Reply:
"What do you mean, 'miss'?"
"I'm sorry, I have a cold"

That is to say, typo

Reply:
So the miss didn't miss a missing m'iss'eaning - OK

Reply:
I still disagree with the implementation of this line, though.
If I want to have a new tileset, I shouldn't be limited to calling it something ending with " Tileset".

Reply:
Let me rephrase - I think this line shouldn't exist.
Permanent visual mods should come through the rulesetMods parameter, above.
Game-specific mods should come through the `mods.addAll(UncivGame.Current.gameInfo.ruleSet.mods)` below.

Reply:
And you're right. Sorry for not marking this back to draft earlier - the longer I think about it the more I tend to think 'I need to _test_ the case a TileSet mod is marked visual but _not_ selected in options'. I'm stalling because I'm not sure how to test properly - that is, which ruleVariants I should look out for...